### PR TITLE
Bug Fixes

### PR DESCRIPTION
**Pull request by NoriteSC**
script and actors not having connect transform in on awake and in on start
with was breaking rigidbody
_this change can create some unitencinal bugs because of how long engine was written with out this change_

join auto anchor having wrong anchor because was including scale

fixed selection do to changes to transform accessibility ?

Edit: this was a glitch ? Source/Editor/Viewport/EditorViewport.cs was reverted back to what it was
idk how i managed to break it
but look like is working again ¯\_(ツ)_/¯

### CODE REVIEW

**Comment by mafiesto4** [comment]
I like the idea of setting prefab location after deserialization but before initialization, so actors can do a setup at the desired transform.
But that scene loading change regarding `OnTransformChanged` event call location might break some stuff as lots of actors use it to update bounds/etc.

**Comment by NoriteSC** [comment]
@mafiesto4 
> OnTransformChanged event call location might break some stuff
i know this but physic need to know the bound before OnStart or is breaking because of scale
for example when prefab is spawned
colliders are not having correct scale on backend because bounds was not correct when colider was created
and not having access to correct transform in Awake/Start also is a problem
because is limiting acces to transform to only Update
and in update transform and bounds are correct

**Comment by mafiesto4** [comment]
Okay, your explanation makes sense to me. Let's target `1.7` branch as it's breaking change and I'll take care of merging and testing this.

File: Source/Engine/Networking/NetworkReplicator.cpp [hunk 1]
```diff
@@ -1738,7 +1738,7 @@ void NetworkInternal::OnNetworkMessageObjectSpawn(NetworkEvent& event, NetworkCl
                 NETWORK_REPLICATOR_LOG(Error, "[NetworkReplicator] Failed to find prefab {}", msgData.PrefabId.ToString());
                 return;
             }
-            prefabInstance = PrefabManager::SpawnPrefab(prefab, nullptr, nullptr);
+            prefabInstance = PrefabManager::SpawnPrefab(prefab, Transform::Identity, nullptr, nullptr);
```
[the remark this answers is not in the document]
Comment (mafiesto4) [reply Source/Engine/Networking/NetworkReplicator.cpp:1741]: That's fine, networking doesn't send transform in spawn message so we don't get proper data here (maybe in the future).

**Pull request merged**